architecture.md

```diff
@@ -0,0 +1,65 @@
+  - A general subgraph that can aggregate all lending data is a beautiful thing. In data science there is an term EXTRACT, TRANSFORM & LOAD (ETL). A general subgraph removes the EXTRACT and TRANSFORM and leaves just the LOAD (i.e. writing queries and loading into their DB) for a data analyst to do.
+  - It is likely that a Subgraph of this size will take +2 weeks to sync. But when the [Fire Hose](https://github.com/streamingfast/firehose) upgrade is ready, this would be a perfect subgraph to test the limits of the indexing speed. 
+  - We should be thinking about this in terms of the The Graph Decentralized Network. It is good for multiple reasons:
+    - Core teams benefit by being able to test the economics of query fees on the decentralized network, test the indexing speed against a massive subgraph, and marketing for the protocol at large.
+    - Indexers lives are easier as they only need to index 1 Subgraph. More queries for a single subgraph. Less transactions and mental overhead for the indexers (ETH is expensive for them to stake on a subgraph).
+    - It makes curators lives easier as well as they need to discover less subgraphs.
+
+The script should simply be a typescript file which combines all manifests into a single manifest, and deploys the subgraph. It will be executed with a node script in `package.json`.
+
+Note - ultimately, as one big subgraph or as many small subgraphs - it is a decision of how to split up the indexing work. Example:
+- If you create one big subgraph, you do all the indexing work ahead of time, and you require the Subgraph Developer to build the Subgraph to include all protocols.
+- If you create many small subgraphs, you parallelize the indexing, but push the work out to Indexers, Curators, and Data Analysts to determine how to aggregate all the data. 
+
+In both cases, queries must be written by the consumers in the end. Neither is right or wrong. What ultimately decides the usefulness of a Subgraph is how many queries it receives in the decentralized network.
+
+## What should the base schema be for the lending protocols?
+Some basic guidelines:
+- It should be simple enough for a traditional finance data scientist to read the schema and query it and fully understand what is going on. No blockchain experience required.
+- But it should be descriptive enough so that someone can get useful information out of it. Hence why it revolves around lending, and we need to standardize lending nomenclature.
+
+Other than that, we need to come to consensus on the fields and the entities. We need to consider it from all angles. The existing schema is just a starting point. Input from all teams will be valuable. This is quite a subjective process, so let's hear each other out and consider all opinions.
+
+## How to create Peripheral subgraphs
+I have some current thoughts, but I am open to other suggestions:
+- Each protocol has it's own folder in `/subgraphs`. 
```

Comment:
Depending on how much is contained in the base schema, we might want to approach this differently.

This would make sense if the base schema covers most of data / use cases. Otherwise we'd end up with a lot of forks, each containing a set of different fields, which defeats the purpose of the standard.

I'd consider one of two options:

1. Have the base schema have enough fields that it could cover as many use cases as possible. This would make the peripheral subgraphs for very specific (and hopefully rare) purposes. In that case, it would make sense for these peripheral subgraphs to only live in forks.
2. Make the subgraphs "modular" so that both implementers and users can pick and choose what they need, with each peripheral subgraph follow a standard. For example, we can have the following:
    - Lending Core: contains the core fields of a lending protocol (e.g. core parameters, lending pools, etc)
    - Lending Usage: contains protocol usage metrics like DAU, total unique users, transaction count, etc
    - Lending Financials: contains protocol revenue, fees, etc
    - Lending Governance: contains governance info

In the second scenario, we probably want the peripheral contracts to live in the main repo instead of the fork, which makes it easier to find and use.

Reply:
Yes I see what you are saying. 

I think both of the options you laid out make sense. Let's see where we get in the next few weeks working on this. I am currently unsure which is the better choice. I think it may reveal itself as we start to develop it.